src/Microsoft.Health.Dicom.Api/Features/Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -160,6 +166,12 @@ private static bool IsTaskCanceledException(Exception ex)
         return ex is TaskCanceledException || (ex is AggregateException aggEx && aggEx.InnerExceptions.Any(x => x is TaskCanceledException));
     }
 
+    private static bool IsCMKException(Exception ex)
+    {
+        return ex is SqlException sqlEx && sqlEx.Number is 40981 or 33183 or 33184 or 40925 ||
+            ex is RequestFailedException rfEx && rfEx.ErrorCode == "KeyVaultEncryptionKeyNotFound";
```

Comment:
Created consts & added a comment about where the error codes come from. 

I don't know if theres a great way to test the SQL side - SqlException is a sealed class with no public constructor, so I'd have to do some ugly reflection stuff to get an instance of it to be able to add it to the unit tests. I didn't add it so far because I'm not sure its worth it, since we would just be injecting the #s anyways.

I was thinking of updating the CMK e2e test to call dicom while its in a bad state (part of the test is disabling the key) and making sure we get back the correct error code. That wouldn't test each SQL number, but it would be something